include/beast/core/detail/sha1.hpp

```diff
@@ -275,15 +275,14 @@ finish(sha1_context& ctx, void* digest) noexcept
     std::uint64_t total_bits =
         (ctx.blocks*64 + ctx.buflen) * 8;
     // pad
-    auto const buflen = ctx.buflen;
     ctx.buf[ctx.buflen++] = 0x80;
+    auto const buflen = ctx.buflen;
```

Comment:
Instead of saving the old buflen here, how about a bool that saves if we should do the extra transform or not?: `auto const extra_transform = (ctx.buflen > BLOCK_BYTES - 8);`